src/drivers/net/bnxt/bnxt.h

```diff
@@ -933,6 +933,19 @@ struct bnxt {
 	void                      *hwrm_addr_req;
 	void                      *hwrm_addr_resp;
 	void                      *hwrm_addr_dma;
+	struct dma_device         *dma;
+	struct dma_mapping        req_mapping;
+	struct dma_mapping        resp_mapping;
+	struct dma_mapping        dma_mapped;
+	struct dma_mapping        tx_mapping;
+	struct dma_mapping        rx_mapping;
+	struct dma_mapping        cq_mapping;
+	struct dma_mapping        nq_mapping;
+	dma_addr_t                tx_addr_mapping;
```

Comment:
There's no real need for these `dma_addr_t` fields, or for having `dma_addr_t` defined as a type specific to the Broadcom driver.  Since you're cleaning up the DMA code anyway, I would suggest deleting the `dma_addr_t` typedef in `bnxt.h`, removing all of these fields within `struct bnxt`, and instead using `dma()` at the point of use for each of these values.

Reply:
Ack, removed this and created macros to do the dma() calls for better readability